src/Components/Block.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace Erusev\Parsedown\Components;
+
+use Erusev\Parsedown\Component;
+use Erusev\Parsedown\Parsing\Context;
+use Erusev\Parsedown\State;
+
+interface Block extends Component
+{
+    /**
+     * @param Context $Context
+     * @param Block|null $Block
+     * @param State|null $State
+     * @return static|null
+     */
+    public static function build(
+        Context $Context,
+        Block $Block = null,
+        State $State = null
```

Comment:
Most blocks don't use state, and the ones that do need to account for it being null. Ideally we should force state to be passed when it is used for parsing, and not require it otherwise.